packages/react/src/input/Input.tsx

```diff
@@ -0,0 +1,50 @@
+import clsx from "clsx";
+import { type ComponentPropsWithRef, forwardRef } from "react";
+
+import { Box } from "../box";
+import { extractSprinkles } from "../sprinkles";
+import { type ExtendProps } from "../utils";
+import { type InpuVariants, input } from "./Input.css";
+
+type InputProps = ExtendProps<
+  ComponentPropsWithRef<"input">,
+  ComponentPropsWithRef<typeof Box>,
+  { isDisabled?: boolean; isInvalid?: boolean } & InpuVariants
+>;
+
+export const Input = forwardRef<HTMLInputElement, InputProps>(
+  (
+    {
+      className,
+      isDisabled,
+      isInvalid,
+      placeholder,
+      size = "md",
+      type,
+      variant = "default",
+      ...props
+    },
+    ref,
+  ) => {
+    const { restProps, sprinkleProps } = extractSprinkles(props);
```

Comment:
Is this needed?

Reply:
Yes Vai We will give the sprinkle props to the Box and pass rest of the props to the input elements like `onChange` etc.

Reply:
Doesn't it happen automatically?

```diff
diff --git a/packages/react/src/input/Input.tsx b/packages/react/src/input/Input.tsx
index d9dd2dd..0754f0a 100644
--- a/packages/react/src/input/Input.tsx
+++ b/packages/react/src/input/Input.tsx
@@ -2,7 +2,6 @@ import clsx from "clsx";
 import { type ComponentPropsWithRef, forwardRef } from "react";
 
 import { Box } from "../box";
-import { extractSprinkles } from "../sprinkles";
 import { type ExtendProps } from "../utils";
 import { type InpuVariants, input } from "./Input.css";
 
@@ -26,22 +25,15 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(
     },
     ref,
   ) => {
-    const { restProps, sprinkleProps } = extractSprinkles(props);
     return (
       <Box
         aria-disabled={isDisabled}
         aria-invalid={isInvalid}
         asChild
-        className={clsx(
-          input({
-            size,
-            variant,
-          }),
-          className,
-        )}
-        {...sprinkleProps}
+        className={clsx(input({ size, variant }), className)}
+        {...props}
       >
-        <input placeholder={placeholder} ref={ref} type={type} {...restProps} />
+        <input placeholder={placeholder} ref={ref} type={type} />
       </Box>
     );
   },
```

this is working for me.

Reply:
It happens automatically now but once we add the wrapper to accommodate the left/right sections it won’t work anymore since we’ll add an extra wrapper

Reply:
@JafarSaifullah